server/src/main/java/org/apache/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -249,6 +278,9 @@ private Thread createDelayedCloseExecutor()
                   dueToShutdownRequest = true;
                 }
               }
+              // This is not possible unless there are bugs in the code of EventReceiverFirehose. AssertionError could
+              // have been thrown instead, but it doesn't seem to make a lot of sense in a background thread. Instead,
+              // we long the error and continue a loop after some pause.
               if (closeTimeNs == null) {
                 log.error(
                     "A bug in EventReceiverFirehose code, "
```

Comment:
`A bug in EventReceiverFirehose code, ` this part of the log message seems to be redundant.

Reply:
It specifies that the following part is not something that a user can affect, but rather that he should report that to Druid developers. Reworded to make it more explicit.